contracts/external/cw-token-swap/src/types.rs

```diff
@@ -0,0 +1,420 @@
+use cosmwasm_schema::cw_serde;
+use cosmwasm_std::{to_binary, Addr, BankMsg, Binary, Coin, CosmosMsg, Deps, Uint128, WasmMsg};
+
+use crate::ContractError;
+
+// We create empty trait and implement it for the types we want to use
+// so we can create a generic function that only accepts a vec of those types
+trait CompletionMsgsToCosmosMsg {
+    fn into_checked_cosmos_msg(
+        self,
+        deps: Deps,
+        denom: &str,
+    ) -> Result<(Uint128, CosmosMsg), ContractError>;
+}
+
+/// This function returns a vector of CosmosMsgs from the given vector of CompletionMsgs
+/// It verifies the msgs is not empty (else return empty vec)
+/// It verifies the total amount of funds matches the funds sent in all messages
+fn completion_to_cosmos_msgs<T: CompletionMsgsToCosmosMsg>(
+    deps: Deps,
+    msgs: Vec<T>,
+    amount: Uint128,
+    string: &str, // extra data we need (denom or contract addr)
```

Comment:
is there a way that we could make this an enum or otherwise avoid overloading the `&str` type? see later comment about not using traits which may help with this.

Reply:
Removed this function, unneeded.